### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
# Blog App - processing data in models
> ***In this feature update:***
- [x] Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Make sure that correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) is used.
- [x] Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Create models
- [x] Use models to insert data
- [x] Create custom methods

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :likes
+  has_many :comments
+
+  def update_posts_counter
+    post.increment!(:PostsCounter)
```
Comment (dasileker) [inline app/models/post.rb]: - [x] Hello @Chimwemwe-127001 nice work ut you did a typo error here at line 7 for the increment, you should use the same as your database. `:Posts_Counter`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  end
+
+  def five_recent_comments
+    comments.order(created_at: :desc).limit(5)
```
Comment (dasileker) [inline app/models/post.rb]: - [x] Also in line 11 there is a typo error(created_at: :desc) please correcte it.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (dasileker) [inline app/models/user.rb]: - [x] I did look into your `DB` folder and they are a bit different, you are not referencing the `Author` and it should refer to class_name user.

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :likes
```
Comment (dasileker) [inline app/models/post.rb]: - [x] The same here goes for the reference to the `Author` with class_name to the user.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Chimwemwe-127001,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

 - Nice PR 👌.
 - Linters are passing 👌.
 - The readme file is detailed  👌.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (Chimwemwe-127001) [reply app/models/post.rb]: Thanks for spotting that

Comment (Chimwemwe-127001) [reply app/models/post.rb]: Done

File: app/models/user.rb (hunk 3 above)
Comment (Chimwemwe-127001) [reply app/models/user.rb]: Done

File: app/models/post.rb (hunk 2 above)
Comment (Chimwemwe-127001) [reply app/models/post.rb]: Done

**Comment by Chimwemwe-127001** [comment]
> ***Thanks, @dasileker, for the insightful review. Surely picked up some valuable lessons from this.***
> ***I have made changes as requested***

**Review by t-yanick — APPROVED** [review]
Hi @Chimwemwe-127001, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chimwemwe-127001** [comment]
> ***Thanks :smiley:***

**Pull request merged**